internal/domain/users/usersStore.go

```
@@ -378,3 +406,90 @@ func (u *usersStore) InvalidateToken(ctx context.Context, accessToken string, us
 
 	return nil
 }
+
+// ExceedsLoginAttempts checks if the user's failed login attempts exceed the maximum allowed attempts
+func (u *usersStore) ExceedsLoginAttempts(ctx context.Context, userID int32) (bool, error) {
+	var count int
+	query := fmt.Sprintf(
+		`select count(*)
+		 from user_login_attempts
+		 where user_id = $1 
+		 and is_successful = false
+		 and created_at >= timezone('UTC', now()) - interval '%v hours';`,
+		BlockLoginHours,
+	)
+	err := u.db.QueryRowxContext(ctx, query, userID).Scan(&count)
+	if err != nil {
+		return true, pdmerr.NewParent(err, "UsersStore: failed to validate login attempts - failed to get recrods count from db")
+	}
+
+	if count >= MaxFailedLoginAttempts {
+		return true, pdmerr.New("UsersStore: failed to validate login attempts - maximum allowed failure login attempts have been reached")
```

return true, nil.

We have the bool return type to indicate whether it exceeds or not, we don't need to return an error for it.

---

fix